src/core/stream/representation/get_needed_segments.ts

```diff
@@ -36,13 +37,14 @@ const { CONTENT_REPLACEMENT_PADDING,
         MAX_TIME_MISSING_FROM_COMPLETE_SEGMENT,
         MINIMUM_SEGMENT_SIZE } = config;
 
+
+type ISegmentMixedWithContent = IContentContext & {segment: ISegment};
```

Comment:
I find the name unclear.

What do you think of `ISegmentContext` (and `IRepresentationContext` instead of `IContentContext`)?